### ISSUE TITLE
False positive for CA1001

### ISSUE THREAD

**Issue by JimBobSquarePants** [dotnet/roslyn-analyzers#6151]
### Analyzer

**Diagnostic ID**: [CA1001](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1001): `Types that own disposable fields should be disposable`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 6 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.400](https://dotnet.microsoft.com/download/dotnet/6.0)

_OR_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 6.0.0 (Latest)

### Describe the bug

I'm seeing several instances in my codebase where `CA1001` is incorrectly triggered. 

- All parent types are structs.
- All implement `IDisposable`
- All errors refer to the same type `EuclideanPixelMap<TPixel>`.

```bash
Error	CA1001	Type 'DitherProcessor' owns disposable field(s) 'pixelMap' but is not disposable	ImageSharp (net6.0)	C:\Users\james\Documents\GitHub\SixLabors\ImageSharp\src\ImageSharp\Processing\Processors\Dithering\PaletteDitherProcessor{TPixel}.cs	77	Active
```

### Steps To Reproduce

The simplest reproduction I have is the following:

```c#
internal struct Invalid : IDisposable
{
    private readonly Stream stream;

    public Invalid(int length) => this.stream = new MemoryStream(length);

    public void Dispose() => this.stream.Dispose();
}
```

Using a class does not trigger the issue.

```c#
internal class Valid : IDisposable
{
    private readonly Stream stream;

    public Valid(int length) => this.stream = new MemoryStream(length);

    public void Dispose() => this.stream.Dispose();
}
```

<img alt="image" src="https://user-images.githubusercontent.com/385879/188642510-5c5e4267-81eb-4bf0-82e4-9fbdf2ebfa4d.png">

A more real world example is the following.

```c#
/// <summary>
/// Used to allow inlining of calls to
/// <see cref="IPaletteDitherImageProcessor{TPixel}.GetPaletteColor(TPixel)"/>.
/// </summary>
/// <remarks>Internal for AOT</remarks>
internal readonly struct DitherProcessor : IPaletteDitherImageProcessor<TPixel>, IDisposable
{
    private readonly EuclideanPixelMap<TPixel> pixelMap;

    [MethodImpl(InliningOptions.ShortMethod)]
    public DitherProcessor(
        Configuration configuration,
        ReadOnlyMemory<TPixel> palette,
        float ditherScale)
    {
        this.Configuration = configuration;
        this.pixelMap = new EuclideanPixelMap<TPixel>(configuration, palette);
        this.Palette = palette;
        this.DitherScale = ditherScale;
    }

    public Configuration Configuration { get; }

    public ReadOnlyMemory<TPixel> Palette { get; }

    public float DitherScale { get; }

    [MethodImpl(InliningOptions.ShortMethod)]
    public TPixel GetPaletteColor(TPixel color)
    {
        this.pixelMap.GetClosestColor(color, out TPixel match);
        return match;
    }

    public void Dispose() => this.pixelMap.Dispose();
}
```

### Expected behavior

The warning should not be triggered

### Actual behavior

The warning is triuggered.

## Additional context
NA

**Comment by myblindy** [comment]
This happens because `struct` types aren't considered disposable, even if they implement `IDisposable`. My PR above relaxes the disposable check rule to allow them as well.

**Comment by JimBobSquarePants** [comment]
@myblindy It looks like there's some feedback on your PR that's blocking merging.
